### PR TITLE
Event dispatcher

### LINKED ISSUE #14045: [Feature]: Move admin menu to topbar
### Describe the feature

Move the admin menu sections to the site topbar

image

### Is your feature request related to a problem?

1. The admin menu items drop to the bottom of the page (not part of the hamburger on mobile)
2. The menu can be very long so you need to scroll a lot to access your menu item

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jdalsem**
feat(admin): moved page menu items to the topbar

fixes #14045

### CODE REVIEW

File: engine/classes/Elgg/Menus/Page.php [hunk 1]
```diff
@@ -464,4 +199,45 @@ public static function registerAvatarEdit(\Elgg\Event $event) {
...
+			
+			$menu_item->setSection('default');
+			if (empty($menu_item->getParentName())) {
+				$menu_item->setParentName($section_name);
+			}
+			
+			elgg_register_menu_item('admin_header', $menu_item);
+			$remove_items[] = $menu_item->getID();
+			
+			elgg_deprecated_notice("The menu item [{$menu_item->getID()}] is using an old section of the admin page menu. These sections have been moved to the admin_header menu. Please update your menu item configuration.", '5.0');
```
Comment (jeabakker) [inline engine/classes/Elgg/Menus/Page.php:235]: `moved to the 'admin_header' menu`

File: engine/classes/Elgg/Menus/Page.php [hunk 2]
```diff
@@ -464,4 +199,45 @@ public static function registerAvatarEdit(\Elgg\Event $event) {
...
+		}
+		
+		foreach ($remove_items as $id) {
+			// can't remove in foreach over MenuItems
```
Comment (jeabakker) [inline engine/classes/Elgg/Menus/Page.php:239]: over?

File: mod/profile/classes/Elgg/Profile/Menus/AdminHeader.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Elgg\Profile\Menus;
+
+/**
+ * Event callbacks for menus
+ *
+ * @since 4.0
```
Comment (jeabakker) [inline mod/profile/classes/Elgg/Profile/Menus/AdminHeader.php:8]: `5.0`

File: views/default/admin/plugins.js [hunk 4]
```diff
@@ -20,30 +20,13 @@ define(['jquery', 'elgg/spinner', 'elgg/Ajax', 'elgg/system_messages', 'elgg/sec
...
-
-						// reapply category filtering
-						$(".elgg-admin-plugins-categories > li.elgg-state-selected > a").trigger('click');
-						initPluginReordering();
-						unfreezePlugins();
-					});
-			}).fail(function() {
-				location.reload();
-			});
+		elgg.forward(this.href);
```
Comment (jeabakker) [inline views/default/admin/plugins.js:29]: why? 

If the event isn't prevented this is not needed, right....

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: views/default/admin/plugins.js (hunk 4 above)
Comment (jdalsem) [reply views/default/admin/plugins.js:29]: totally right

**Review by jeabakker — APPROVED** [review]

File: engine/classes/Elgg/Assets/ImageFetcherService.php [hunk 5]
```diff
@@ -5,7 +5,6 @@
 use Elgg\Cache\SystemCache;
 use Elgg\Config;
 use Elgg\Exceptions\InvalidArgumentException;
-use GuzzleHttp\Client;
 use GuzzleHttp\Cookie\CookieJar;
 use GuzzleHttp\Exception\TransferException;
 use GuzzleHttp\RequestOptions;
```
[the remark this answers is not in the document]
Comment (jdalsem) [reply engine/classes/Elgg/Assets/ImageFetcherService.php:10]: yes

**Pull request merged**